common/src/main/java/net/william278/husksync/redis/RedisManager.java

```diff
@@ -88,18 +90,51 @@ public void initialize() throws IllegalStateException {
         }
 
         // Subscribe using a thread (rather than a task)
+        enabled = true;
         new Thread(this::subscribe, "husksync:redis_subscriber").start();
     }
 
     @Blocking
     private void subscribe() {
-        try (Jedis jedis = jedisPool.getResource()) {
-            jedis.subscribe(
-                    this,
-                    Arrays.stream(RedisMessage.Type.values())
-                            .map(type -> type.getMessageChannel(clusterId))
-                            .toArray(String[]::new)
-            );
+        boolean reconnected = false;
+        while (enabled && !Thread.interrupted() && jedisPool != null && !jedisPool.isClosed()) {
+            try (Jedis jedis = jedisPool.getResource()) {
+                if (reconnected) {
+                    plugin.log(Level.INFO, "Redis connection is alive again");
+                }
+                // Subscribe channels and lock the thread
+                jedis.subscribe(
+                        this,
+                        Arrays.stream(RedisMessage.Type.values())
+                                .map(type -> type.getMessageChannel(clusterId))
+                                .toArray(String[]::new)
+                );
+            } catch (Throwable t) {
+                // Thread was unlocked due error
+                if (enabled) {
+                    if (reconnected) {
+                        plugin.log(Level.WARNING, "Redis connection dropped, automatic reconnection in 8 seconds", t);
```

Comment:
grammar: `Connection to the Redis server was lost. Attempting reconnection in 8 seconds...`